server/src/test/integration/org/apache/cassandra/sidecar/common/DelegateIntegrationTest.java

```diff
@@ -246,7 +271,7 @@ private void validateJmxConnections(VertxTestContext context, Set<Integer> jmxCo
         }
         else if (upInstanceCount == 3)
         {
-            assertThat(jmxConnectedInstances).containsExactly(1, 2, 3);
+            context.verify(() -> assertThat(jmxConnectedInstances).containsExactly(1, 2, 3));
```

Comment:
must wrap in the verify block to correctly fail the test if assertion fails. Otherwise, context/flag are not closed, and test gets stuck until timeout.